executor/cook/executor.py

```diff
@@ -184,7 +186,7 @@ def kill_task(process, shutdown_grace_period_ms):
     -------
     Nothing
     """
-    shutdown_grace_period_ms = max(shutdown_grace_period_ms - 100, 0)
+    shutdown_grace_period_ms = max(shutdown_grace_period_ms - (1000 * cook.TERMINATE_GRACE_SECS), 0)
     if is_process_running(process):
         logging.info('Waiting up to {} ms for process to terminate'.format(shutdown_grace_period_ms))
         process.terminate()
```

Comment:
Since we're launching the sub-process with shell=True here, these signals are actually only being sent to the wrapper shell used to launch our executor command; i.e., they never actually make it to the executor<sup>[[1]](https://stackoverflow.com/q/4789837/1427124)</sup>. I think your approach using `os.killpg` is definitely the right approach, but we should also be using it everywhere that we currently call `process.terminate()` or `process.kill()` to ensure that the signals actually reach the true executor's process.

We can verify that the executor command is in fact receiving the `SIGTERM` using a command like this in the integration tests:
```
bash -c 'function handle_term { echo GOT TERM SIGNAL; }; trap handle_term SIGTERM; sleep 100'
```

Reply:
Done.